SKFlatMaker/src/SKFlatMaker.cc

```diff
@@ -2622,15 +2629,60 @@ void SKFlatMaker::fillJet(const edm::Event &iEvent)
     bool tightLepVetoJetID = (NHF<0.90 && NEMF<0.90 && NumConst>1 && MUF<0.8) && ((fabs(eta)<=2.4 && CHF>0 && CHM>0 && CEMF<0.80) || fabs(eta)>2.4) && fabs(eta)<=2.7;
-    if(fabs(eta)>3.0){
-      tightJetID = (NEMF<0.90 && NHF>0.02 && NumNeutralParticles>10) && fabs(eta)>3.0;
-      tightLepVetoJetID = false; //FIXME default is false or true? following CAT2016
+
+    if(DataYear==2016){
+      //=== 2016 jet IDs
+      //==== https://twiki.cern.ch/twiki/bin/view/CMS/JetID13TeVRun2016
+      if(fabs(eta)>3.0){
+	tightJetID = (NEMF<0.90 && NumNeutralParticles>10 );
+	tightLepVetoJetID = (NEMF<0.90 && NumNeutralParticles>10 );
+      }
+      else if(fabs(eta)>2.7){
+	tightJetID = (NHF<0.98 && NEMF>0.01 && NumNeutralParticles>2);
+	tightLepVetoJetID = (NHF<0.98 && NEMF>0.01 && NumNeutralParticles>2);
+      }
+      else {
+	tightJetID        = (NHF<0.90 && NEMF<0.90 && NumConst>1) &&            ((fabs(eta)<=2.4 && CHF>0 && CHM>0 && CEMF<0.99) || fabs(eta)>2.4) && fabs(eta)<=2.7;
+	tightLepVetoJetID = (NHF<0.90 && NEMF<0.90 && NumConst>1 && MUF<0.8) && ((fabs(eta)<=2.4 && CHF>0 && CHM>0 && CEMF<0.90) || fabs(eta)>2.4) && fabs(eta)<=2.7;
+      }
     }
-    else if(fabs(eta)>2.7){
-      tightJetID = (NEMF>0.02 && NEMF<0.99) && (NumNeutralParticles>2) && abs(eta)>2.7 && abs(eta)<=3.0;
-      tightLepVetoJetID = false; //FIXME default is false or true? following CAT2016
+    else   if(DataYear<=2017){
+      //=== 2017 jet IDs                                                                                                                                                                               
+      //==== https://twiki.cern.ch/twiki/bin/view/CMS/JetID13TeVRun2017                                                                                                                                    
+      if(fabs(eta)>3.0){
+        tightJetID = (NEMF<0.90 && NHF>0.02 && NumNeutralParticles>10);                                                                                                                                   
+        tightLepVetoJetID = (NEMF<0.90 && NHF>0.02 && NumNeutralParticles>10);
+      }
+      else if(fabs(eta)>2.7){
+        tightJetID = (NEMF>0.02 && NEMF<0.99) && (NumNeutralParticles>2);                                                                                                                                 
+        tightLepVetoJetID =  (NEMF>0.02 && NEMF<0.99) && (NumNeutralParticles>2);
+      }
+      else{
+	tightJetID        = (NHF<0.90 && NEMF<0.90 && NumConst>1) &&            ((fabs(eta)<=2.4 && CHF>0 && CHM>0)              || fabs(eta)>2.4) && fabs(eta)<=2.7;
+	tightLepVetoJetID = (NHF<0.90 && NEMF<0.90 && NumConst>1 && MUF<0.8) && ((fabs(eta)<=2.4 && CHF>0 && CHM>0 && CEMF<0.80) || fabs(eta)>2.4) && fabs(eta)<=2.7;	
+      }
     }
-
+    else{
```

Comment:
```suggestion
    else if(DataYear==2018){
```
For the safety reason, I think we'd better specify the year and add
`else{`
`    cout << "DataYear is not set : DataYear = " << DataYear << endl;`
`    exit(EXIT_FAILURE);`
`}`
after this statement

Reply:
Made this change in new PR